JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -371,27 +359,57 @@ private fun TabWithTopics(
  */
 @Composable
 private fun TabWithSections(
-    sections: TopicsMap,
+    sections: List<InterestSection>,
     selectedTopics: Set<TopicSelection>,
     onTopicSelect: (TopicSelection) -> Unit
 ) {
-    LazyColumn(Modifier.navigationBarsPadding()) {
-        sections.forEach { (section, topics) ->
-            item {
-                Text(
-                    text = section,
-                    modifier = Modifier
-                        .padding(16.dp)
-                        .semantics { heading() },
-                    style = MaterialTheme.typography.subtitle1
-                )
+    BoxWithConstraints {
+        val windowSize = remember(maxWidth) { getWindowSize(maxWidth) }
+        val columns = remember(windowSize) { if (windowSize == WindowSize.Compact) 1 else 2 }
+
+        val itemMaxWidth = rememberItemMaxWidth(windowMaxWidth = maxWidth, columns = columns)
+        // Group interest sections given the number of columns that fill the screen.
+        // Sections are distributed evenly on the screen using the modulus operator.
+        val groupedSections: Array<MutableList<InterestSection>> = remember(columns) {
```

Comment:
I think this can be simplified as is pretty easily:

```
        val groupedSections = sections
            .withIndex()
            .groupBy({ it.index % columns }) { it.value }
            .values
```